server/src/main/java/com/objectcomputing/checkins/services/checkindocument/CheckinDocumentController.java

```diff
@@ -64,13 +80,18 @@
      * @return {@link HttpResponse<CheckinDocument>}
      */
 
-    @Post(value = "/")
-    public HttpResponse<CheckinDocument> createACheckinDocument(@Body @Valid CheckinDocumentCreateDTO checkinDocument,
-                                                                HttpRequest<CheckinDocumentCreateDTO> request) {
-        CheckinDocument newCheckinDocument = checkinDocumentService.save(new CheckinDocument(checkinDocument.getCheckinsId(), checkinDocument.getUploadDocId()));
-        return HttpResponse
-                .created(newCheckinDocument)
-                .headers(headers -> headers.location(URI.create(String.format("%s/%s", request.getUri(), newCheckinDocument.getId()))));
+    @Post("/")
+    public Single<HttpResponse<CheckinDocument>> createCheckinDocument(@Body @Valid CheckinDocumentCreateDTO checkinDocument,
+                                                                    HttpRequest<CheckinDocumentCreateDTO> request) {
+        return Single.fromCallable(() -> checkinDocumentService.save(new CheckinDocument(checkinDocument.getCheckinsId(),checkinDocument.getUploadDocId())))
+                .observeOn(Schedulers.from(eventLoopGroup))
+                .map(createdCheckinDocument -> {
+                    //Using code block rather than lambda so we can log what thread we're in
```

Comment:
This comment was there for demonstration purposes. If you get rid of it you can change this to a lambda and the result will be a much cleaner read.

Reply:
updated.